contracts/seed/Seed.sol

```diff
@@ -113,11 +128,36 @@ contract Seed {
             _individualCap <= _classCap && _classCap <= hardCap,
             "Seed: caps are invalid"
         );
+        require(!closed, "Seed: should not be closed");
+        require(_classCap > 0, "Seed: class Cap should be bigger then 0");
+        _;
+    }
+
+    modifier classBatchRestrictions(
+        bytes32[] memory _classNames,
+        uint256[] memory _classCaps,
+        uint256[] memory _individualCaps,
+        uint256[] memory _vestingCliffs,
+        uint256[] memory _vestingDurations,
+        address[][] memory _allowlist
+    ) {
+        uint256 arrayLength = _classNames.length;
         require(
-            block.timestamp < startTime,
-            "Seed: vesting is already started"
+            arrayLength == _classCaps.length &&
+                arrayLength == _individualCaps.length &&
+                arrayLength == _vestingCliffs.length &&
+                arrayLength == _vestingDurations.length &&
+                arrayLength == _allowlist.length,
+            "Seed: All provided arrays should be same size"
+        );
+        require(
+            arrayLength <= 100,
+            "Seed: Can't add batch with more then 100 classes"
+        );
+        require(
+            classes.length + arrayLength < 256,
+            "Seed: can't add more then 256 classes"
```

Comment:
You are correct, the comments state something different. I will change it to 256

Reply:
Removed the class length check in `contributorClassRestriction` because we only going to add classes through the `addClassesAndAllowlists` function, which checks in the `classBatchRestrictions` for the length